### ISSUE TITLE
Drop sync API execution for GraphCall

### ISSUE THREAD

**Issue by sav007** [Shopify/mobile-buy-sdk-android#428]
We don't need to support both async and sync API:
`com.shopify.buy3.GraphCall#execute` and `com.shopify.buy3.GraphCall#enqueue(com.shopify.buy3.GraphCall.Callback<T>)`.
We should consider of having only one async.

**Comment by markj** [comment]
There are times a sync API is really helpful..

E.g. within an Android service where you want to run synchronously but you're not blocking the main thread.

**Comment by sav007** [comment]
That's true that sync API helps a lot in some cases but one issue is that sync API can't provide multiple results, for now all MBSDK calls are one shot call, but if in future we need to return result more then once (good example return cache version first and then hit network and return network response) we have to use callback. And in this case we have inconsistent API: with sync `execute` we can get only one response, with `enqueue` we can get more than one. It will increase of effort to maintain such API
